### PR TITLE
Invite Links: Implements service layer methods.

### PR DESCRIPTION
**Pull request by aerych**
Refs #15356 

This PR adds service layer methods supporting the Invite Links feature.  It depends on related changes in WordPress-Kit: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/356.  

While the methods are added they are not currently being called.  This will happen in a follow up PR that implements the new UI for P2 sites.

To test:

You can test service methods individually via the following snippets when applied to InvitePersonViewController.viewDidLoad. 

To test generating invites:
```
        let service = PeopleService(blog: blog, context: context)
        service?.generateInviteLinks(blog.dotComID!.intValue, success: { links in
            DDLogDebug("\(links)")
        }, failure: { error in
            DDLogDebug("\(error)")
        })
```
To test fetching invites:
```
        let service = PeopleService(blog: blog, context: context)
        service?.fetchInviteLinks(blog.dotComID!.intValue, success: { links in
            DDLogDebug("\(links)")
        }, failure: { error in
            DDLogDebug("\(error)")
        })
```
To test disabling invites: 
```
        let service = PeopleService(blog: blog, context: context)
        service?.disableInviteLinks(blog.dotComID!.intValue, success: {
            DDLogDebug("Success")
        }, failure: { error in
            DDLogDebug("\(error)")
        })
```

@ScoutHarris, would you kindly? 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Services/PeopleService.swift [hunk 1]
```diff
@@ -263,9 +263,199 @@ struct PeopleService {
...
 }
 
+// MARK: - Invite Links Related
+
+extension PeopleService {
+
+    /// Convenience method for retrieving invite links from core data.
+    ///
+    /// - Parameters:
+    ///   - siteID: The ID of the site.  The site should be a P2.
```
Comment (ScoutHarris) [inline WordPress/Classes/Services/PeopleService.swift]: > The site should be a P2.

Should it? I know we'll add the restriction in the UI. But fetching, generating, and disabling invites all work for a non-P2 site. And there aren't checks in any of these methods to force a P2. I just find these comments a bit misleading/confusing.

**Review by ScoutHarris — APPROVED** [review]
Just a questions about comments. Otherwise, 👍 !

:shipit:

File: WordPress/Classes/Services/PeopleService.swift (hunk 1 above)
Comment (aerych) [reply WordPress/Classes/Services/PeopleService.swift]: Good points. I'm game to drop that comment.

**Comment by aerych** [comment]
Thanks @ScoutHarris!  I needed to make [one more change](https://github.com/wordpress-mobile/WordPress-iOS/pull/15970/commits/059cf12b7d151fd29b9f5aa21bcd980de5005262) to account for another PR that merged before this one.  Game for one final quick check for paranoia sake?

**Comment by ScoutHarris** [comment]
I saw you sneak that in. 😉 Looks good to me, especially since it removes a force unwrap!

**Comment by aerych** [comment]
I'd forgotten about that force unwrap until I rewrote that bit of code. It was never meant for production 😬.  Glad it got cut in the end.

**Pull request merged**